### PR TITLE
refactored type init

### LINKED ISSUE #199: make srlinux default type to be ixr6

### PR DESCRIPTION
**Pull request by hellt**
close #199

### CODE REVIEW

File: clab/config.go [hunk 1]
```diff
@@ -226,11 +226,22 @@ func (c *cLab) groupInitialization(nodeCfg *NodeConfig, kind string) string {
...
 		return nodeCfg.Type
+	case c.Config.Topology.Kinds[nodeCfg.Kind].Type != "":
+		return c.Config.Topology.Kinds[nodeCfg.Kind].Type
+	case c.Config.Topology.Defaults.Type != "":
+		return c.Config.Topology.Defaults.Type
+	}
+	// default type if not defined
+	switch nodeCfg.Kind {
+	case "srl":
+		return "ixr6"
```
Comment (steiler) [inline clab/config.go:242]: This should maybe not be a hardcoded value but rather reference the srlTypes map.
My approach considered the first key in the map.

Comment (hellt) [reply clab/config.go:242]: I dont think its a good idea. The map might be shuffled by someone without noticing that its values are used in something like default type

Comment (steiler) [reply clab/config.go:242]: I just do not like arbitrary strings that much. I'd rather opt for an enum kind of type.
If someone messes with that string it breaks and if ixr6 is entirely removed from the list there is inconsistency as well.
But ok.

Comment (hellt) [reply clab/config.go]: I made it a constant, so that this will be visible better 53c7b4c

Comment (hellt) [reply clab/config.go]: btw, range over dict will yield inconsistent results, as map access is random

**Pull request merged**